scripts/shared/lib/clusters_kind

```diff
@@ -264,11 +259,47 @@ function download_ovnk() {
     )
 }
 
+function prepare_ovn_ic() {
+    echo "Building ovn-kubernetes with interconnect (OVN-IC) from source"
+    rm -rf ovn-kubernetes || true
+    git clone https://github.com/ovn-org/ovn-kubernetes
+    pushd ovn-kubernetes || exit
+
+    pushd go-controller || exit
+    make
+    popd || exit
+
+    cp go-controller/_output/go/bin/* dist/images
+
+    pushd dist/images || exit
+    echo "ref: $(git rev-parse --symbolic-full-name HEAD) commit: $(git rev-parse HEAD)" > git_info
+    docker build -t "${OVN_IMAGE}" -f Dockerfile.fedora .
+    docker push "${OVN_IMAGE}"
+    popd || exit
+
+    popd || exit
```

Comment:
```
In scripts/shared/lib/clusters_kind line 250:
        cd ovn-kubernetes
        ^---------------^ SC2164 (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

Did you mean: 
        cd ovn-kubernetes || exit


In scripts/shared/lib/clusters_kind line 269:
    pushd ovn-kubernetes
    ^------------------^ SC2164 (warning): Use 'pushd ... || exit' or 'pushd ... || return' in case pushd fails.

Did you mean: 
    pushd ovn-kubernetes || exit


In scripts/shared/lib/clusters_kind line 279:
    popd
    ^--^ SC2164 (warning): Use 'popd ... || exit' or 'popd ... || return' in case popd fails.

Did you mean: 
    popd || exit

For more information:
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... |...
```

https://github.com/submariner-io/shipyard/actions/runs/5868225584/job/15910567495?pr=1338